tensorflow/dtensor/mlir/tests/dtensor_reduce_scatter_lowering.mlir

```diff
@@ -33,6 +33,48 @@ func.func @lower_reduce_scatter_sum_cpu() -> (tensor<2048x8192xf32>) {
   func.return %3: tensor<2048x8192xf32>
 }
 
+// CHECK-LABEL: func @lower_reduce_scatter_gpu_mesh
+func.func @lower_reduce_scatter_gpu_mesh(%arg0: tensor<i32>,
```

Comment:
Does this test require setting the GPU NCCL environment? (Line 1 was not changed).

Reply:
The default value of the env var is on, but I can add the var here to be more explicit anyway and incase we want to change it to off by default.

Reply:
Oh I didn't notice you changed the default of the C++ environment parsing. The current default is to not enable NCCL, since TF's Virtual GPU feature doesn't (fully) work with NCCL (for the collectives). Virtual GPU is how we demo DTensor in a single physical GPU notebook.

Later maybe we can upgrade this to automatically enable NCCL when no TF vGPU is used, but let's don't change the wide behavior yet with this PR.

Reply:
Oops, I didn't realize python had it off by default. Thanks, let me fix that.